### ISSUE TITLE
Lint for highlighting possibly invalid coercion `Box<dyn Any>` -> `&dyn Any` with implicit deref

### ISSUE THREAD

**Issue by Ved-s** [rust-lang/rust-clippy#13004]
### What it does

Give a warning when doing possibly unwanted &dyn Any coercion from a Box<dyn Any>

### Advantage

- Less headache figuring out why the Any is of wrong type

### Example

```rust

let b: Box<dyn Any> = ...;

// What I'd write to get a reference to the inner Any
// but instead i get a reference to the box itself
let r: &dyn Any = &b; // lint: possibly unwanted coercion

// Suggest replacing with:
let r: &dyn Any = b.deref();
```
